### PR TITLE
Expose websockets on api prefix v2

### PR DESCRIPTION
**Pull request by jamesmarshall24**
##### SUMMARY

Expose extended endpoint for `api/controller/v2/websocket/`

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - API

### CODE REVIEW

File: awx/main/routing.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ async def __call__(self, scope, receive, send):
 
 
 websocket_urlpatterns = [
+    re_path(r'api/controller/v2/websocket/$', consumers.EventConsumer.as_asgi()),
```
Comment (dsavineau) [inline awx/main/routing.py]: not sure if we want to hardcode `controller` here or reuse the `OPTIONAL_API_URLPATTERN_PREFIX` variable from the settings to build this path

https://github.com/ansible/awx/blob/devel/awx/settings/defaults.py

Comment (jamesmarshall24) [reply awx/main/routing.py]: that would also require a change to line 63?

File: tools/docker-compose/ansible/roles/sources/templates/nginx.locations.conf.j2 [hunk 2]
```diff
@@ -10,7 +10,7 @@ location {{ (ingress_path + '/favicon.ico').replace('//', '/') }} {
     alias /awx_devel/awx/public/static/favicon.ico;
 }
 
-location ~ ^({{ (ingress_path + '/websocket/').replace('//', '/') }}|{{ (ingress_path + '/api/websocket/').replace('//', '/') }}) {
+location ~ ^({{ (ingress_path + '/websocket/').replace('//', '/') }}|{{ (ingress_path + '/api/websocket/').replace('//', '/') }}|{{ (ingress_path + '/api/controller/v2/websocket/').replace('//', '/') }}) {
```
Comment (chrismeyersfsu) [inline tools/docker-compose/ansible/roles/sources/templates/nginx.locations.conf.j2]: We don't want to hard code in `controller` into the awx project. 
We will need to create a new playbook variable and use it here. That same variable should also set `settings.OPTIONAL_API_URLPATTERN_PREFIX = {{ some_playbook_var }}`

Comment (jamesmarshall24) [reply tools/docker-compose/ansible/roles/sources/templates/nginx.locations.conf.j2]: @chrismeyersfsu where exactly? I am not familiar with your docker-compose tools and I'm unsure where its best appropriate to add this variable.

**Review by chrismeyersfsu — APPROVED** [review]

**Review by dsavineau — APPROVED** [review]

**Pull request merged**